### ISSUE TITLE
Available for sale ?

### ISSUE THREAD

**Issue by 0mars** [newdigate/teensy-eurorack#12]
Is it available for sale as a module ? looks interesting

**Comment by newdigate** [comment]
Hopefully soon... Im working on building some documentation to help with the build, if you wanted to try to build yourself...  
I'd like to start to sell kits for DIY assembly and fully assembled modules soon.

**Comment by newdigate** [comment]
Would you be interested joining a mailing list for those interested when getting notified about when this module is available for purchase as a kit, assembled?

**Comment by nicolasdf1** [comment]
Yes! Please add me to a mailing list. Thanks for this project and your
dedication to creating documentation and also offering it as DIY.
The power of the DIY community is unmeasureable.
All the best,
Nicolas Fellas

**Comment by jonasfehr** [comment]
Hi, please add me to the mailing list too... 
Kind regards 
Jonas

fehrjonas at gmail com

**Comment by jonasfehr** [comment]
I have had good success, having china soldering all SMD parts. Maybe this could be an option for the kit versions?

**Comment by newdigate** [comment]
@simonredfern added to the list... 

https://github.com/simonredfern

**Comment by zazze** [comment]
Yes, please add me to the mailing list!
I am definitely interested.
Thank you for sharing your work!
Best wishes
Johan Zetterqvist
    Den torsdag 12 mars 2020 10:14:48 CET, Nic Newdigate <notifications@github.com> skrev:  
 
 


Would you be interested joining a mailing list for those interested when getting notified about when this module is available for purchase as a kit, assembled?

**Comment by newdigate** [comment]
2 done.... 

5 to go.... 

hang in there guys...!!!! I need to test my current revision 1.9 and Im also working on 1.9.1 which will hopefully help us to integrate the ad5754 into the teensy audio library...

**Comment by erwincoumans** [comment]
nice, please sign me up for diy kit, erwin.coumans at gmail.com

**Comment by newdigate** [comment]
added...

**Comment by tompilot** [comment]
Hi, this looks really cool! +1 for the mailing list please. 
Do you have any experience with noise issues? I have an Audio Shield + Teensy 4.1 here, connected the line in/outs to Eurorack and i get really a LOT of noise and hiss. Power supply is coming from Eurorack also. I wonder how it performs with different (better) ADC/DACs and on a real PCB instead of breadboard.

**Comment by newdigate** [comment]
Hi Tom. 

Breadboards are very good for audio, they are usually quite intermittent, you can get them to work, but they are very fragile, I experienced this first hand. You should download Kicad and see if you can build a circuit and a pcboard.

Other thing is that I assume your not sending control voltage into the audio card. CV is +- 12v (24 volts) and line audio is +-1V so you will either kill your audio codec or you'll get clipping, Im assuming you already know this..  

I doubt either of these things will cause a hiss thou.

What you really want to do is hook your MCLK, BCLK, LRCLK and TX to a logic analyser like Saleae and check that it can decode and that you are not sending junk to your audio codec. 

Also make sure that vcc is not too far above the i2s levels, when vcc is higher that signals, it becomes more suseptable to noise.

**Comment by newdigate** [comment]
I think your hiss is coming because your audio levels are really low and youre amplifying the shit out of it? That sounds like a source of hiss..

**Comment by newdigate** [comment]
@tompilot if you like you can continue this conversation here [![Gitter](https://badges.gitter.im/newdigate/teensy-eurorack.svg)](https://gitter.im/newdigate/teensy-eurorack?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)

**Comment by tompilot** [comment]
Thank you Nic for your detailed answer! I have done some experimentation and finally found some sort of "solution". 

My current DSP approach is a lot about FIR filtering and convolution of long impulse responses (which is all quite CPU hungry). I figured out that when using long IRs with the Teensy Audio Lib the I2S clock starts to jitter (Teensy is I2S master). Also, some of the noise came from putting the power supply through the breadboard. 

To get a clean environment i did: 
- Use another audio shield (WM8731) with crystal and **use I2S Slave mode**, so that the crystal creates the clock signal - not the Teensy. I also noticed changes in the noise when other CPU- or timer consuming tasks are running, in my case printing on a display and using encoders. Probably the SGT5000 codec would also be doing fine in slave mode, but the PJRC Audio shield does not support a crystal on it. 
- Solder the line input on the development board instead of going through the breadboard (blocking cap and two resistors in case of WM8731). Having it on the breadboard was very sensitive to picking up noise. 
- Solder the power supply directly to the Teensy. I am using the 5V line of my Rack as the power. 
- In general: Disable the 5V supply from the USB programming cable by cutting a trace on the Teensy (you should never use two power sources at once)

By now, i only have the I2S and i2c lines going over the breadboard, which does not seem to have an effect. 
Will switch over to gitter now, but i just wanted to give my reply here for completeness :-)

**Comment by newdigate** [comment]
Just a quick update - if you want to follow the project and get updates, priority pre-orders, discounts, pcbs -  please consider supporting this project in patreon: https://www.patreon.com/teensy_eurorack

Thanks :)
